detectors/resources/README.md

```diff
@@ -1,5 +1,7 @@
 # Resource Detectors for OpenTelemetry
 
+> NOTE: This artifact has moved to [Opentelemetry Java Contrib](https://github.com/open-telemetry/opentelemetry-java-contrib) repo. All future releases/updates made to this artifact would be managed in the new repository. The code for this artifact is likely to be deleted from here in the near future. 
```

Comment:
This PR actually removes this package as an artifact, so even if I update the code to add `@Deprecated` annotations, the artifact won't be released.

I can add the module back so that it is included in the next release with the deprecation warning in the code.

Reply:
If a user references it in their code, will it continue to pull in the previous version of the resource detector?  Will they get notified that it disappeared from our repo in any way?

Reply:
In case we - 
 - Add `resource-detector` module back to the code and add `@Deprecated` annotations.
     - The user will be able to pull in the detector code from this repo and see a deprecation warning.
     - We will need to support the detector in this repo for another release and keep it in sync with the upstream one. 
     
- Do not add `resource-detector` module back and add `@Deprecated` annotations.
    - For the next release, when the users try to update dependencies, they will not be able to download v0.28.0 for the `resource-detector` module since it would have moved. 
    - The `@Deprecated` annotation would not have the intended effect since a dependency version containing these annotations would never have been released.
    
In both cases, the users may continue to use current version - v0.27.0 - which contains the detector code from this repo.

Reply:
In the case we don't add the module back, would the user ever find out that we removed it?  Would they notice when they try to update other dependencies?

Reply:
Not explicitly, since the build tool would simply complain dependency - `com.google.cloud.opentelemetry:detector-resources:0.28.0-alpha` not found.

The way upstream does is deprecates the artifact and then removes the code in a separate release. Reference - 
Moving semconv in OTel Java 
  - semconv was deprecated in [v1.30.0](https://github.com/open-telemetry/opentelemetry-java/releases/tag/v1.30.0) was removed in [v1.31.0](https://github.com/open-telemetry/opentelemetry-java/releases/tag/v1.31.0).
  
 We could do it similarly.

Reply:
I would prefer that.  Are there type aliases in java, or a similar way for us to just re-expose the upstream detector?

Reply:
Ok, I will make the necessary changes. 👍🏻 

I dont think there are type aliases in Java. The closest I think is type inheritance. 

IIUC, the intention here is to just alias the current files to the upstream detector.

I think we should not do that - its best if we maintain the code in a single place.